In [1]:
import pandas as pd 
import matplotlib.pyplot as plt
import seaborn as sns
%matplotlib inline 

sns.set_style('darkgrid')
plt.rcParams['font.size'] = 14
plt.rcParams['figure.figsize'] = (9,5)
plt.rcParams['figure.facecolor'] = '#00000000'

In [2]:
election_df = pd.read_csv('1976_2020_president.csv')

In [3]:
election_df

year    state state_po  state_fips  state_cen  state_ic        office  \
0     1976  ALABAMA       AL           1         63        41  US PRESIDENT   
1     1976  ALABAMA       AL           1         63        41  US PRESIDENT   
2     1976  ALABAMA       AL           1         63        41  US PRESIDENT   
3     1976  ALABAMA       AL           1         63        41  US PRESIDENT   
4     1976  ALABAMA       AL           1         63        41  US PRESIDENT   
...    ...      ...      ...         ...        ...       ...           ...   
4282  2020  WYOMING       WY          56         83        68  US PRESIDENT   
4283  2020  WYOMING       WY          56         83        68  US PRESIDENT   
4284  2020  WYOMING       WY          56         83        68  US PRESIDENT   
4285  2020  WYOMING       WY          56         83        68  US PRESIDENT   
4286  2020  WYOMING       WY          56         83        68  US PRESIDENT   

                    candidate              party_detailed writein  \
0               CARTER, JIMMY                    DEMOCRAT   False   
1                FORD, GERALD                  REPUBLICAN   False   
2              MADDOX, LESTER  AMERICAN INDEPENDENT PARTY   False   
3     BUBAR, BENJAMIN ""BEN""                 PROHIBITION   False   
4                   HALL, GUS         COMMUNIST PARTY USE   False   
...                       ...                         ...     ...   
4282            JORGENSEN, JO                 LIBERTARIAN   False   
4283            PIERCE, BROCK                 INDEPENDENT   False   
4284                      NaN                         NaN    True   
4285                OVERVOTES                         NaN   False   
4286               UNDERVOTES                         NaN   False   

      candidatevotes  totalvotes   version  notes party_simplified  
0             659170     1182850  20210113    NaN         DEMOCRAT  
1             504070     1182850  20210113    NaN       REPUBLICAN  
2               9198     1182850  20210113    NaN            OTHER  
3               6669     1182850  20210113    NaN            OTHER  
4               1954     1182850  20210113    NaN            OTHER  
...              ...         ...       ...    ...              ...  
4282            5768      278503  20210113    NaN      LIBERTARIAN  
4283            2208      278503  20210113    NaN            OTHER  
4284            1739      278503  20210113    NaN            OTHER  
4285             279      278503  20210113    NaN            OTHER  
4286            1459      278503  20210113    NaN            OTHER  

[4287 rows x 15 columns]

# Checking the NaN values

In [4]:
election_df.isna().sum()

year                   0
state                  0
state_po               0
state_fips             0
state_cen              0
state_ic               0
office                 0
candidate            287
party_detailed       456
writein                3
candidatevotes         0
totalvotes             0
version                0
notes               4287
party_simplified       0
dtype: int64

As we have total records 4287 in the dataset and in the notes column there are also 4287 NaN values.
it indicates that we do not need notes column. 
Office and version has the same value all along. 
so we can drop notes, office and version columns.

In [5]:
election_df.drop(['notes', 'office', 'version'], axis = 1, inplace=True)

In [6]:
election_df.isna().sum()

year                  0
state                 0
state_po              0
state_fips            0
state_cen             0
state_ic              0
candidate           287
party_detailed      456
writein               3
candidatevotes        0
totalvotes            0
party_simplified      0
dtype: int64

In [7]:
election_df['party_detailed'].fillna("Unknown", inplace=True)

In [8]:
election_df.dropna(subset = ['writein'], inplace = True)

In [9]:
election_df.info()

<class 'pandas.core.frame.DataFrame'>
Index: 4284 entries, 0 to 4286
Data columns (total 12 columns):
 #   Column            Non-Null Count  Dtype 
---  ------            --------------  ----- 
 0   year              4284 non-null   int64 
 1   state             4284 non-null   object
 2   state_po          4284 non-null   object
 3   state_fips        4284 non-null   int64 
 4   state_cen         4284 non-null   int64 
 5   state_ic          4284 non-null   int64 
 6   candidate         3998 non-null   object
 7   party_detailed    4284 non-null   object
 8   writein           4284 non-null   object
 9   candidatevotes    4284 non-null   int64 
 10  totalvotes        4284 non-null   int64 
 11  party_simplified  4284 non-null   object
dtypes: int64(6), object(6)
memory usage: 435.1+ KB
